### PR TITLE
[WIP] Merge API

### LINKED ISSUE #10: Merge functionality
The goal is to add merging capability to repofs, in comparing branches and making merge commits.

- [x] Compare branches for changes (no deep diffing)
- [x] Committing with more than one parent
- [ ] Replace the existing merge method

### PR DESCRIPTION
**Pull request by Soreine**
Fix #10

### CODE REVIEW

**Comment by SamyPesse** [comment]
The integration with the editor can maybe pass by an event:

```js
fs.on('conflicts.resolve.needed', function(conflicts, next) {
    // conflicts.head = { sha: "<sha>" }
    // conflicts.base = { sha: "<sha>" }
    // conflicts.tree = { ... }

    // the editor shows the dialog and resolve the conflicts here, then call (async):

    next(null, conflictsResolved);

    // or in case or error:

    next(err);
});
```

It will allow in `repofs` to handle conflicts whenever they are (during a commit, pull, push, etc):

```js
Fs.prototype.resolveConflicts = function(conflicts) {
    var d = Q.defer();

    var next = function(err, solved) {
        if (err) d.reject(err);
        else d.resolve(solved);
    }

    if (this.listenerCount('conflicts.resolve.needed') == 0) {
        next(new Error('Couldn\'t resolve conflicts'));
    } else {
        this.emit('conflicts.resolve.needed', conflicts, next);
    }

    return d.promise;
}
```


For example after a `.commit`:

```
....
.fail(function(err) {
     if (err.code == 409) {
           return that.fetchConflicts(...)
           .then(that.resolveConflicts);
    } else {
           throw err;
    }
});
```

File: .eslintrc [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
     "rules": {
-        "indent": [ 2, 4 ],
```
Comment (SamyPesse) [inline .eslintrc]: indentation

File: README.md [hunk 2]
```diff
@@ -158,7 +158,7 @@ fs.move('README.txt', 'README2.txt').then(function() { ... });
 fs.mvdir('lib', 'lib2').then(function() { ... });
 ```
 
-##### Working with branches
+##### Manipulating with branches
```
Comment (SamyPesse) [inline README.md]: typo

**Comment by SamyPesse** [comment]
We'll also need a `conflicts.resolved` event, triggered after result of `next` has been merged.

File: scripts/test-uhub.sh [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+touch README.md
+git add README.md
+git commit -m "Initial commit"
+cd ../..
+
+REPO_PATH=$(pwd)/.tmp/repo/
+REPO_BASE64=$(printf $REPO_PATH | base64)
+
+# Start uhub
+./uhub --mode=absolute --port=127.0.0.1:6666 > /dev/null  &
```
Comment (AaronO) [inline scripts/test-uhub.sh]: You should use `--mode=single`. It means all repo IDs will resolve to that repo path.

Pass `--mode=single` and `--root=PATH_TO_REPO`

Comment (SamyPesse) [inline scripts/test-uhub.sh]: Thanks :+1:

**Pull request merged**